fuzz/fuzz_targets/ruff_parse_simple.rs

```diff
@@ -4,13 +4,59 @@
 #![no_main]
 
 use libfuzzer_sys::{fuzz_target, Corpus};
-use ruff_python_ast::source_code::round_trip;
+use ruff_python_ast::prelude::{lexer, Mode, Parse, ParseError, Suite};
+use ruff_python_ast::source_code::{Generator, Locator, Stylist};
 
 fn do_fuzz(case: &[u8]) -> Corpus {
     let Ok(code) = std::str::from_utf8(case) else { return Corpus::Reject; };
 
     // just round-trip it once to trigger both parse and unparse
-    let _ = round_trip(code, "fuzzed-source.py");
+    let locator = Locator::new(code);
+    let python_ast = match Suite::parse(code, "fuzzed-source.py") {
+        Ok(stmts) => stmts,
+        Err(ParseError { offset, .. }) => {
+            let offset = offset.to_usize();
+            assert!(
+                code.is_char_boundary(offset),
+                "Invalid error location {} (not at char boundary)",
+                offset
+            );
+            return Corpus::Keep;
+        }
+    };
+
+    let tokens: Vec<_> = lexer::lex(code, Mode::Module).collect();
+
+    for maybe_token in tokens.iter() {
+        match maybe_token.as_ref() {
+            Ok((_, range)) => {
+                let start = range.start().to_usize();
+                let end = range.end().to_usize();
```

Comment:
Do you mind elaborating a bit on what problem this is solving? Just trying to catch up on the relevant issue. It looks like we're detecting here whether the lexer is emitting tokens.

Reply:
Yes, and checking if the indices specified by those tokens are aligned on character boundaries. This was not the case for #5004 and led to panics in some cases.

Reply:
As for why you'd do this: you can use this as a bit of an indirect way to detect miscomputed locations, as the fuzzer has the potential to inject the UTF-8 bytes at the miscomputed indices.